### PR TITLE
Make platformvm implement `block.HeightIndexedChainVM`

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Implements `block.HeightIndexedChainVM` on the platformvm to kick off the P-chain proposervm height index (https://github.com/ava-labs/avalanchego/blob/master/vms/proposervm/vm.go#L341)

## How this works

Implement interface

## How this was tested

CI

### CODE REVIEW

File: vms/platformvm/state/state.go [hunk 1]
```diff
@@ -704,6 +708,14 @@ func (s *state) donePrune() error {
 	return s.singletonDB.Put(prunedKey, nil)
 }
 
+func (s *state) VerifyHeightIndex() error {
+	shouldPrune, err := s.shouldPrune()
+	if err != nil || shouldPrune {
+		return block.ErrIndexIncomplete
+	}
+	return nil
```
Comment (abi87) [inline vms/platformvm/state/state.go:716]: nit: this implementation will change post 1.11.x right? It'll be return nil

**Review by abi87 — APPROVED** [review]
nits

File: vms/platformvm/state/state.go [hunk 2]
```diff
@@ -164,6 +165,9 @@ type State interface {
 	// TODO: remove after v1.11.x is activated
 	PruneAndIndex(sync.Locker, logging.Logger) error
 
+	// TODO: remove after v1.11.x is activated
```
[the remark this answers is not in the document]
Comment (dhrubabasu) [reply vms/platformvm/state/state.go:168]: It's not - we'll remove this function from this file and just return `nil` directly in `vm.go`

File: vms/platformvm/state/state.go (hunk 1 above)
Comment (dhrubabasu) [reply vms/platformvm/state/state.go:716]: Correct. We'll modify `vm.go` to return `nil` and remove this function.

File: vms/platformvm/vm.go [hunk 3]
```diff
@@ -90,6 +91,9 @@ type VM struct {
 
 	txBuilder txbuilder.Builder
 	manager   blockexecutor.Manager
+
+	// TODO: Remove after v1.11.x is activated
+	pruned bool
```
Comment (StephenButtolph) [inline vms/platformvm/vm.go:96]: needs to be atomic

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**